pageserver/src/virtual_file.rs

```diff
@@ -61,6 +63,176 @@ pub(crate) mod owned_buffers_io {
     }
 }
 
+#[derive(Debug)]
+pub enum VirtualFile {
+    Buffered(VirtualFileInner),
```

Comment:
I wonder if it would make more sense to have a `kind: VirtualFileKind` member together with `inner: VirtualFileInner`?

of course, it depends on your plans for the future, but that way, the `inner()`, `inner_mut()` etc functions could be simplified a little.

Reply:
I made the change in 6d03e2810d66b2a4cf2e8390cc5cb7037174335b. The mode is currently actually only used when opening the virtual file. It would become useful when read/write start consuming it.